src/lib/wallet/index.ts

```diff
@@ -105,33 +108,60 @@ export default class Wallet extends Web3Wallet implements HederaNativeWallet {
     id: number
     topic: string
     chainId: HederaChainId
-    accountId: AccountId
     method: HederaJsonRpcMethod
-    body: Transaction
+    transaction?: Transaction
+    query?: Query<any>
+    requestParams: any
+    accountId?: AccountId
   } {
     const { id, topic } = event
     const {
-      request: { method, params },
+      request: { method, params: requestParams },
       chainId,
     } = event.params
+    
+
+    let transaction: Transaction | undefined;
+
+    const methodsWithoutTransaction = [
+      HederaJsonRpcMethod.SendTransactionOnly,
+      HederaJsonRpcMethod.SignTransactionAndSend,
+      HederaJsonRpcMethod.SignTransactionBody,
+      HederaJsonRpcMethod.SignTransactionAndSend,
+    ];
+
+    if (methodsWithoutTransaction.includes(method as HederaJsonRpcMethod)) {
+      transaction = base64StringToTransaction(requestParams[0]);
+    }
+
+    const accountId = transaction?.transactionId?.accountId!;
+
+    let query: Query<any> | undefined;
```

Comment:
nice!  I noticed query required a type parameter as well...in contrast to `Transaction`